test/over_react/component_declaration/component_base_test.dart

```diff
@@ -1040,6 +1040,58 @@ main() {
       });
     });
 
+    group('UiStatefulComponent2', () {
+      TestStatefulComponent2Component statefulComponent;
+
+      setUp(() {
+        statefulComponent = new TestStatefulComponent2Component();
+        statefulComponent.state = JsBackedMap.from({'test': true});
+      });
+
+      group('`state`', () {
+        group('getter:', () {
+          test('returns a UiState view into the component\'s state map', () {
+            expect(statefulComponent.state, const TypeMatcher<TestStatefulComponent2State>());
+
+            expect(statefulComponent.state, isNot(same(statefulComponent.state.state)));
+
+            statefulComponent.state.state['testKey'] = 'testValue';
+            expect(statefulComponent.state, containsPair('testKey', 'testValue'));
+          });
+
+          test('caches the UiState object for the same map', () {
+            var state1 = statefulComponent.state;
+            var state2 = statefulComponent.state;
+            expect(state1, same(state2));
+          });
+
+          test('creates a new UiState object when the state map changes', () {
+            var stateBeforeChange = statefulComponent.state;
+            statefulComponent.state = JsBackedMap.from({'test': true});
+            var stateAfterChange = statefulComponent.state;
+
+            expect(stateBeforeChange, isNot(same(stateAfterChange)));
+          });
+        });
+
+        group('setter:', () {
+          test('sets the unwrapped Map, as react-dart requires it to', () {
+            var newBackingMap = JsBackedMap.from({'test': true});
+            statefulComponent.state = newBackingMap;
+            expect(statefulComponent.state.state, same(newBackingMap));
+          });
+        });
+      });
+
+      test('newState() returns a new UiProps instance backed by a new Map', () {
+        var newState1 = statefulComponent.newState();
+        var newState2 = statefulComponent.newState();
+        expect(newState1, const TypeMatcher<TestStatefulComponent2State>());
+        expect(newState2, const TypeMatcher<TestStatefulComponent2State>());
```

Comment:
I feel like we're using `isA<...>()` elsewhere in Dart2-only libs since it is the recommended replacement for `const isInstanceOf<...>()`.